packages/cw1155/src/receiver.rs

```diff
@@ -0,0 +1,73 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{to_binary, Binary, CosmosMsg, HumanAddr, StdResult, Uint128, WasmMsg};
+
+use crate::msg::TokenId;
+
+/// Cw1155ReceiveMsg should be de/serialized under `Receive()` variant in a HandleMsg
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub struct Cw1155ReceiveMsg {
+    pub operator: HumanAddr,
+    pub from: Option<HumanAddr>,
```

Comment:
From should always be set, right? Otherwise you don't know who is calling this.

What is the different between from and operator?

Reply:
When from is `None`, it means new tokens is minted directly to the receiver by operator.